tests/Faithlife.Analyzers.Tests/NullTestTernariesTests.cs

```diff
@@ -0,0 +1,102 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Framework;
+
+namespace Faithlife.Analyzers.Tests
+{
+	[TestFixture]
+	public class NullTestTernariesTests : CodeFixVerifier
+	{
+		[TestCase("Person person = null;", "var firstName = person?.FirstName;")]
+		[TestCase("var person = new Person { FirstName = \"Bob\", LastName = \"Dole\" };", "var firstName = person.Age?.BirthYear;")]
+		public void ValidUsage(string declaration, string ternary)
+		{
+			string validProgram = $@"
+namespace TestApplication
+{{
+	public class Person
+	{{
+		public string FirstName {{ get; set; }}
+		public string LastName {{ get; set; }}
+		public Age? Age {{ get; set; }}
+	}}
+
+	public struct Age
+	{{
+		public int? InYears {{ get; set; }}
+
+		public int? BirthYear 
+		{{
+			get
+			{{
+				return System.DateTime.Now.Year - InYears;
+			}}
+		}}
+	}}
+
+	internal static class TestClass
+	{{
+		public static void UtilityMethod()
+		{{
+			{declaration}
+			{ternary}
+		}}
+	}}
+}}";
+
+			VerifyCSharpDiagnostic(validProgram);
+		}
+
+		[TestCase("var firstName = person != null ? person.FirstName : null;")]
+		[TestCase("var firstName = person == null ? null : person.FirstName;")]
```

Comment:
These are already handled by IDE0031; we shouldn't create (or maintain) functionality that duplicates built-in analysers.